src/modules/Dashboard/features/Applet/Popups/DeletePopup/DeletePopup.tsx

```diff
@@ -56,6 +56,21 @@ export const DeletePopup = ({ onCloseCallback, 'data-testid': dataTestid }: Dele
     deletePopupClose();
   };
 
+  useEffect(() => {
+    if (activeModal === Modals.Confirmation) {
```

Comment:
let's get rid of `useEffect` that could be replaced with a callback. it's a way more intuitive and much improve readability of the code, simplify writing the tests and debugging.